GoRogue/MapGeneration/Region.cs

```diff
@@ -0,0 +1,374 @@
+        /// <param name="imposing">the region to check for common outer points</param>
+        public void RemoveOverlappingOuterpoints(Region imposing)
+        {
+            foreach (Point c in imposing.OuterPoints)
+            {
+                while (OuterPoints.Contains(c))
+                    ((List<Point>)OuterPoints).Remove(c);
+                while (InnerPoints.Contains(c))
+                    ((List<Point>) InnerPoints).Remove(c);
+            }
+        }
+
+        /// <summary>
+        /// Removes (from this region) the inner points shared in common
+        /// with the imposing region
+        /// </summary>
+        /// <param name="imposing">the region to check for common outer points</param>
+        public void RemoveOverlappingInnerpoints(Region imposing)
+        {
+            foreach (Point c in imposing.InnerPoints)
+            {
+                while (OuterPoints.Contains(c))
+                    ((List<Point>) OuterPoints).Remove(c);
+                while (InnerPoints.Contains(c))
+                    ((List<Point>) InnerPoints).Remove(c);
+            }
+        }
+        /// <summary>
+        /// Removes (from this region) the outer points shared in common
+        /// with the imposing region
+        /// </summary>
+        /// <param name="imposing">the region to check for common outer points</param>
+        public void RemoveOverlappingPoints(Region imposing)
+        {
+            foreach (Point c in imposing.OuterPoints.Concat(imposing.InnerPoints))
+            {
+                while (OuterPoints.Contains(c))
+                    ((List<Point>) OuterPoints).Remove(c);
+                while (InnerPoints.Contains(c))
+                    ((List<Point>) InnerPoints).Remove(c);
+            }
+        }
+
+        /// <summary>
+        /// Rotates this region by an arbitrary number of degrees
+        /// </summary>
+        /// <param name="degrees">The amount of degrees to rotate this region</param>
+        /// <param name="doToSelf">Whether to perform this rotation on this region, or return a new region</param>
+        /// <param name="origin">The Point around which to rotate</param>
+        /// <returns>This region, rotated</returns>
+        /// <remarks>
+        /// This is destructive to the region's Connections, so you should try to refrain from generating those
+        /// until after you've performed your rotations.
+        /// </remarks>
+        public virtual Region Rotate(float degrees, bool doToSelf, Point origin = default)
+        {
+            Point center;
+            Region Region;
+            int quarterTurns = 0;
+            double radians = SadRogue.Primitives.MathHelpers.ToRadian(degrees);
+
+            List<Point> corners = new List<Point>();
+            if (origin == default(Point))
+                center = new Point((Left + Right) / 2, (Top + Bottom) / 2);
+            else
+                center = origin;
+
+            while (degrees < 0)
```

Comment:
Could be wrong but I think we can replace both of these 360-degree based while loops with `degrees = MathHelpers.WrapAround(degrees, 360);`